samples/MegaDrone/src/main/cpp/MegaDroneEngine.cpp

```diff
@@ -18,11 +18,12 @@
 #include <memory>
 #include "MegaDroneEngine.h"
 
-MegaDroneEngine::MegaDroneEngine(std::vector<int> cpuIds) : AudioEngine(std::make_shared<DefaultAudioStreamCallback>()) {
-    mDefaultCallbackPtr->setParent(*this);
+MegaDroneEngine::MegaDroneEngine(std::vector<int> cpuIds) {
+    mCallback = std::make_unique<DefaultAudioStreamCallback>(*this);
```

Comment:
The purpose of supplying `*this` to the ctor here is to allow the callback to call its parent's `restart` method when the stream disconnects. Do you think this might be more flexible if instead the callback had a method to allow a function to be specified which would be called on a disconnect event?

Reply:
This callback could take a specific function type (such as `std::function<void()>`) that specified the restart function. Since the function would be a member function, the function would need to be passed `std::bind(MegaDroneEngine::restartStream, this)`.

Reply:
SGTM. Want me to implement this?

Reply:
I will address in a follow up CL